synapse/lib/iq.py

```diff
@@ -689,6 +748,100 @@ def getDmonCore(self):
             core.fini()
             dmon.fini()
+            s_scope.pop('syn:dmon')
+            s_scope.pop('syn:core')
+            s_scope.pop('syn:cmd:core')
+            s_scope.pop('syn:test:link')
+
+    @contextlib.contextmanager
+    def getSslCore(self, conf=None, configure_roles=False):
+        dconf = {'auth:admin': 'root@localhost',
+                 'auth:en': 1, }
+        if conf:
+            conf.update(dconf)
+        conf = dconf
+
+        amesgs = (
+            ('auth:add:user', {'user': 'user@localhost'}),
+            ('auth:add:role', {'role': 'creator'}),
+            ('auth:add:rrule', {'role': 'creator',
+                                'rule': ('node:add',
+                                         {'form': '*'})
+                                }),
+            ('auth:add:rrule', {'role': 'creator',
+                                'rule': ('node:tag:add',
+                                         {'tag': '*'})
+                                }),
+            ('auth:add:rrule', {'role': 'creator',
+                                'rule': ('node:prop:set',
+                                         {'form': '*', 'prop': '*'})
+                                }),
+            ('auth:add:role', {'role': 'deleter'}),
+            ('auth:add:rrule', {'role': 'deleter',
+                                'rule': ('node:del',
+                                         {'form': '*'})
+                                }),
+            ('auth:add:rrule', {'role': 'deleter',
+                                'rule': ('node:del',
+                                         {'form': '*'})
+                                }),
+            ('auth:add:rrule', {'role': 'deleter',
+                                'rule': ('node:tag:del',
+                                         {'tag': '*'})
+                                }),
+            ('auth:add:rrule', {'role': 'deleter',
+                                'rule': ('node:prop:set',
+                                         {'form': '*', 'prop': '*'})
+                                }),
+            ('auth:add:urole', {'user': 'user@localhost', 'role': 'creator'}),
+            ('auth:add:urole', {'user': 'user@localhost', 'role': 'deleter'}),
+        )
+
+        with self.getDirCore(conf=conf) as core:
+            s_scope.set('syn:core', core)
+            dirn = s_scope.get('dirn')
+            writeCerts(dirn)
+            cafile = os.path.join(dirn, 'ca.crt')
+            keyfile = os.path.join(dirn, 'server.key')
+            certfile = os.path.join(dirn, 'server.crt')
+            userkey = os.path.join(dirn, 'user.key')
+            usercrt = os.path.join(dirn, 'user.crt')
+            rootkey = os.path.join(dirn, 'root.key')
+            rootcrt = os.path.join(dirn, 'root.crt')
+            with s_daemon.Daemon() as dmon:
+                s_scope.set('syn:dmon', dmon)
+                dmon.share('core', core)
+                link = dmon.listen('ssl://localhost:0/',
```

Comment:
This will break on OSX, we should fix before merging this PR, as @jnwatson 's PR will cause this to fail in CI.

Reply:
holding till master merge